packages/gatsby-link/src/index.js

```diff
@@ -153,7 +155,11 @@ class GatsbyLink extends React.Component {
             // loaded before continuing.
             if (process.env.NODE_ENV === `production`) {
               e.preventDefault()
-              window.___push(this.state.to)
+              if (window.GATSBY_SW_UPDATED) {
+                window.location = this.state.to
```

Comment:
How about checking this flag and doing location change in https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/production-app.js#L68 (TBD if before or after redirect handling)? So programatic navigation would get same behaviour (if someone manually call `window.___push`/`window.___replace`)

Reply:
oh snap, that is much better! will move this over there